plugin/ale/cursor.vim

```diff
@@ -81,7 +95,9 @@ function! ale#cursor#EchoCursorWarning(...)
     let index = s:BinarySearch(loclist, pos[1], pos[2])
 
     if index >= 0
-        call ale#cursor#TruncatedEcho(loclist[index]['text'])
+        let l = loclist[index]
+        let msg = s:GetMessage(l.linter_name, l.type, l.text)
```

Comment:
It takes more time? It will just be passing a reference to the dictionary.

You can leave it how it is. It's just an observation.